pkg/util/migrations/migrations.go

```diff
@@ -1,19 +1,27 @@
 package migrations
 
 import (
+	"fmt"
+
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/tools/cache"
 	v1 "kubevirt.io/api/core/v1"
 	"kubevirt.io/client-go/log"
 
+	"kubevirt.io/kubevirt/pkg/controller"
 	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
 )
 
 const CancelMigrationFailedVmiNotMigratingErr = "failed to cancel migration - vmi is not migrating"
 
-func ListUnfinishedMigrations(store cache.Store) []*v1.VirtualMachineInstanceMigration {
-	objs := store.List()
-	migrations := []*v1.VirtualMachineInstanceMigration{}
+func ListUnfinishedMigrations(indexer cache.Indexer) []*v1.VirtualMachineInstanceMigration {
+	objs, err := indexer.ByIndex(controller.UnfinishedIndex, controller.UnfinishedIndex)
+	if err != nil {
+		log.Log.Reason(err).Errorf("Failed to use unfinished index")
+		return nil
+	}
+
+	var migrations []*v1.VirtualMachineInstanceMigration
```

Comment:
Is it worth checking `!migration.IsFinal()` again? The indexer already did that, otherwise the migration shouldn't be in the index?